src/MRA.Pages.Api/Controllers/AuthorizationController.cs

```diff
@@ -1,19 +1,43 @@
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authentication;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
 
 namespace MRA.Pages.Api.Controllers;
 
-public class AuthorizationController : Controller
+public class AuthorizationController(IConfiguration configuration) : Controller
 {
-    public IActionResult CallBack(string atoken)
+    public async Task<IActionResult> CallBack(string atoken)
     {
-        HttpContext.Response.Cookies.Append("authToken", atoken, new CookieOptions
+        var tokenHandler = new JwtSecurityTokenHandler();
+        var res = tokenHandler.ValidateToken(atoken, new TokenValidationParameters
         {
-            Secure = false,
-            Expires = DateTimeOffset.Now.AddDays(10),
-            SameSite = SameSiteMode.Lax,
-            HttpOnly = true,
-            IsEssential = false
-        });
+            ValidateIssuerSigningKey = false,
+            IssuerSigningKey =
+                new SymmetricSecurityKey(
+                    Encoding.UTF8.GetBytes(configuration["JWT:Secret"]!)),
+            ValidateIssuer = false,
+            ValidateAudience = false
+        }, out _);
+        var identity = new ClaimsIdentity(res.Claims, CookieAuthenticationDefaults.AuthenticationScheme);
```

Comment:
Make sure that it will not throw 500 error when validation fails. It would be good if you move to another class serivce